### PR TITLE
6298 Error in BoundingBox.Contains(BoundingFrustum)

### PR DESCRIPTION
**Pull request by Tauroctonos**
Fixed logical error in frustum containment method; see #6298

### CODE REVIEW

File: MonoGame.Framework/BoundingBox.cs [hunk 1]
```diff
@@ -132,20 +130,20 @@ public ContainmentType Contains(BoundingFrustum frustum)
                 return ContainmentType.Intersects;
 
 
-            // If we get here, it means the first (and only) point we checked was actually contained in the frustum.
-            // So we assume that all other points will also be contained. If one of the points is disjoint, we can
-            // exit immediately saying that the result is Intersects
+            // If we get here it means the first (and only) point we checked was outside the frustum, so we assume
```
Comment (initram) [inline MonoGame.Framework/BoundingBox.cs:133]: The point we checked here was outside the box, not the frustum.

File: MonoGame.Framework/BoundingBox.cs [hunk 2]
```diff
@@ -132,20 +130,20 @@ public ContainmentType Contains(BoundingFrustum frustum)
...
             for (; i < corners.Length; i++)
             {
                 this.Contains(ref corners[i], out contained);
-                if (contained != ContainmentType.Contains)
+                if (contained == ContainmentType.Contains)
                     return ContainmentType.Intersects;
 
             }
 
             // If we get here, then we know all the points were actually contained, therefore result is Contains
```
Comment (initram) [inline MonoGame.Framework/BoundingBox.cs:145]: This comment should also be updated to something like: If we get here, then we know all the points are actually outside the box, therefore result is Disjoint.

On that note, is it not possible to have an intersection between a BoundingBox and BoundingFrustum where none of their corners are contained within the other volume? The Frustum is not box shaped and is not axis aligned, so a corner could technically pass through the whole of the other volume and not be contained within while still intersecting.

Comment (Tauroctonos) [reply MonoGame.Framework/BoundingBox.cs:145]: I had the same thought, but if the Frustum's bigger than the Box then it can't really be contained by it. Maybe that case should return an Intersect instead though

Comment (Tauroctonos) [reply MonoGame.Framework/BoundingBox.cs:145]: You could have an intersection where the Fustrum's bigger than the box, but if you're calling `.Contains()` on the box while it's entirely contained in the Fustrum it makes sense to still return Intersects I think.
So, I think if all the Fustrum's points are outside the box, we can check to see if the box is contained in the Fustrum, and if we get anything but Disjoint return Intersects.

**Comment by initram** [comment]
Have you considered this edge case? I would think that would result in intersect.
edge case

Here no corner is inside the volume of the other shape, so just checking corners are not enough to handle all intersection cases.

**Comment by Tauroctonos** [comment]
> Have you considered this edge case? I would think that should result in intersect.
> edge case
> 
> Here no corner is inside the volume of the other shape, so just checking corners are not enough to handle all intersection cases.

**Comment by tomspilman** [comment]
This needs a test case for us to merge @Tauroctonos . 

If you or someone adds one we can merge it.

**Comment by SimonDarksideJ** [comment]
@Tauroctonos are you able to provide the Unit test to accompany this PR, or should it be closed?
